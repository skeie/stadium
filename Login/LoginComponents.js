// @flow

import React from 'react';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import { View, StyleSheet, TouchableOpacity } from 'react-native';
import TextInput from '../components/TextInput';
import { Poppins } from '../components/StyledText';
import colors from '../constants/Colors';
import Button from '../components/Button';

const loginText = "Don't have an account? Click here to Signup!";
const signupText = 'Already have an account? Click here to login';

const styles = StyleSheet.create({
    textInput: {
        padding: 10,
        width: '50%',
        borderWidth: 1,
        borderColor: colors.primaryText,
        borderRadius: 10,
    },
});

type Props = {
    isSignup: boolean,
    email: string,
    onInputChange: (string, string) => void,
    username: string,
    password: string,
    onToggleSignup: () => void,
    onButtonPressed: () => void,
    error: string,
};

const LoginComponent = (props: Props) => {
    return (
        <View flex={1} style={{ backgroundColor: colors.primary }}>
            <View justifyContent="center" alignItems="center" flex={1}>
                <MaterialCommunityIcons name="stadium" size={64} color={colors.primaryText} />
                <Poppins type="header">Welcome to Stadiumz</Poppins>
            </View>
            <View flex={1} justifyContent="flex-start" alignItems="center">
                {props.isSignup && (
                    <TextInput
                        spellCheck={false}
                        placeholder="Username"
                        name="username"
                        autoFocus
                        value={props.username}
                        style={styles.textInput}
                        onChangeText={text => {
                            props.onInputChange('username', text);
                        }}
                    />
                )}
                <View padding={5} />
                <TextInput
                    spellCheck={false}
                    value={props.email}
                    style={styles.textInput}
                    placeholder="Email"
                    autoFocus={!props.isSignup}
                    onChangeText={text => {
                        props.onInputChange('email', text);
                    }}
                />
                <View padding={5} />
                <TextInput
                    spellCheck={false}
                    secureTextEntry
                    onChangeText={text => {
                        props.onInputChange('password', text);
                    }}
                    value={props.password}
                    style={styles.textInput}
                    placeholder="Password"
                />
                <View padding={15} />
                <Button style={{ width: '50%' }} onPress={props.onButtonPressed}>
                    {props.isSignup ? 'Signup' : 'Login'}
                </Button>
                <View padding={5} />
                <TouchableOpacity onPress={props.onToggleSignup}>
                    <Poppins>{props.isSignup ? signupText : loginText}</Poppins>
                </TouchableOpacity>
                <View padding={15}>
                    <Poppins type="error">{props.error}</Poppins>
                </View>
            </View>
        </View>
    );
};

export default LoginComponent;
