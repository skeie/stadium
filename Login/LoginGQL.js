import gql from 'graphql-tag';

export const signup = gql`
    mutation($email: String!, $password: String!, $name: String!) {
        signup(email: $email, password: $password, name: $name) {
            token
        }
    }
`;

export const login = gql`
    mutation($email: String!, $password: String!) {
        login(email: $email, password: $password) {
            token
        }
    }
`;
