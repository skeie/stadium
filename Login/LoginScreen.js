// @flow

import React from 'react';
import {
  Image,
  Platform,
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
  TextInput,
} from 'react-native';
// $FlowFixMe
import Login from './LoginContainer';

export default class LoginScreen extends React.Component<*, *> {
  static navigationOptions = {
    header: null,
  };

  handleLogin = () => {
    const { navigate } = this.props.navigation;
    navigate('Main', { showFirstTimeView: true });
  };

  render() {
    return <Login onLogin={this.handleLogin} />;
  }
}
