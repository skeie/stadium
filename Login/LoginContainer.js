// @flow

import React, { Component } from 'react';
import { LayoutAnimation } from 'react-native';
import { login, signup } from './LoginGQL';
import layout from '../constants/Layout';
import Login from './LoginComponents';
// $FlowFixMe
import { graphql, compose } from 'react-apollo';
import { onSignIn } from '../util/auth';
type State = {
    username: string,
    password: string,
    isSignup: boolean,
    email: string,
    error: string,
};

class LoginContainer extends Component<*, State> {
    state = {
        username: '',
        password: '',
        email: '',
        isSignup: true,
        error: '',
    };

    handleInputChange = (name: string, value: string) => {
        this.setState({
            [name]: value,
        });
    };

    handleToggleSignup = () => {
        LayoutAnimation.configureNext(layout.customLayoutLinear);
        this.setState(({ isSignup }) => ({
            isSignup: !isSignup,
        }));
    };

    handleSignup = async () => {
        try {
            const { data } = await this.props.signup({
                variables: {
                    name: this.state.username,
                    password: this.state.password,
                    email: this.state.email,
                },
            });
            const { token } = data.signup;
            await onSignIn(token);
            this.props.onLogin();
        } catch (error) {
            console.log('error handleSignup', error);
            this.setState({
                error: 'A user with this email already exist, maybe its you? Try to login :)',
            });
        }
    };

    handleLogin = async () => {
        try {
            const { data } = await this.props.login({
                variables: {
                    email: this.state.email,
                    password: this.state.password,
                },
            });
            const { token } = data.login;
            await onSignIn(token);
            this.props.onLogin();
        } catch (error) {
            console.log('error handleSignup', error);
            this.setState({
                error: 'A user with this email already exist, maybe its you? Try to login :)',
            });
        }
    };

    handleButtonPressed = () => {
        this.state.isSignup ? this.handleSignup() : this.handleLogin();
    };

    render() {
        return (
            <Login
                {...this.state}
                onButtonPressed={this.handleButtonPressed}
                onInputChange={this.handleInputChange}
                onToggleSignup={this.handleToggleSignup}
            />
        );
    }
}

export default compose(
    graphql(login, {
        name: 'login',
    }),
    graphql(signup, {
        name: 'signup',
    }),
)(LoginContainer);
