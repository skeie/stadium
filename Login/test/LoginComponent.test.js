/* eslint flowtype/require-valid-file-annotation: 0 */
/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import { Text } from 'react-native';
import Login from '../LoginComponents'

const onInputChange = jest.fn();

describe('Login component', () => {
    it('render login layout', () => {
        const wrapper = shallow((
            <Login isSignup={false} onInputChange={onInputChange} />
        ));
        expect(wrapper).toMatchSnapshot();
    })
    it('render signup layout', () => {
        const wrapper = shallow((
            <Login isSignup onInputChange={onInputChange} />
        ));
        expect(wrapper).toMatchSnapshot();
    })
})
