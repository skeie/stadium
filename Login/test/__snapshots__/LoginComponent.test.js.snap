// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component render login layout 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  keyboardVerticalOffset={0}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ADF1D2",
      "flex": 1,
      "paddingTop": 20,
    }
  }
>
  <ScrollViewMock
    contentContainerStyle={
      Object {
        "flexGrow": 1,
        "justifyContent": "center",
      }
    }
    keyboardShouldPersistTaps="handled"
    style={
      Object {
        "backgroundColor": "#ADF1D2",
      }
    }
  >
    <View
      alignItems="center"
      flex={1}
      justifyContent="center"
    >
      <Icon
        allowFontScaling={false}
        color="#553555"
        name="stadium"
        size={64}
      />
      <Poppins
        type="header"
      >
        Welcome to Stadiumz
      </Poppins>
    </View>
    <View
      alignItems="center"
      flex={1}
      justifyContent="center"
    >
      <View
        padding={5}
      />
      <Input
        autoFocus={true}
        onChangeText={[Function]}
        placeholder="Email"
        style={
          Object {
            "borderColor": "#553555",
            "borderRadius": 10,
            "borderWidth": 1,
            "padding": 10,
            "width": "50%",
          }
        }
      />
      <View
        padding={5}
      />
      <Input
        onChangeText={[Function]}
        placeholder="Password"
        secureTextEntry={true}
        style={
          Object {
            "borderColor": "#553555",
            "borderRadius": 10,
            "borderWidth": 1,
            "padding": 10,
            "width": "50%",
          }
        }
      />
      <View
        padding={15}
      />
      <Button
        style={
          Object {
            "width": "50%",
          }
        }
      >
        Login
      </Button>
      <View
        padding={5}
      />
      <TouchableOpacity
        activeOpacity={0.2}
      >
        <Poppins>
          Don't have an account? Click here to Signup!
        </Poppins>
      </TouchableOpacity>
    </View>
  </ScrollViewMock>
</KeyboardAvoidingView>
`;

exports[`Login component render signup layout 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  keyboardVerticalOffset={0}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ADF1D2",
      "flex": 1,
      "paddingTop": 20,
    }
  }
>
  <ScrollViewMock
    contentContainerStyle={
      Object {
        "flexGrow": 1,
        "justifyContent": "center",
      }
    }
    keyboardShouldPersistTaps="handled"
    style={
      Object {
        "backgroundColor": "#ADF1D2",
      }
    }
  >
    <View
      alignItems="center"
      flex={1}
      justifyContent="center"
    >
      <Icon
        allowFontScaling={false}
        color="#553555"
        name="stadium"
        size={64}
      />
      <Poppins
        type="header"
      >
        Welcome to Stadiumz
      </Poppins>
    </View>
    <View
      alignItems="center"
      flex={1}
      justifyContent="center"
    >
      <Input
        autoFocus={true}
        name="username"
        onChangeText={[Function]}
        placeholder="Username"
        style={
          Object {
            "borderColor": "#553555",
            "borderRadius": 10,
            "borderWidth": 1,
            "padding": 10,
            "width": "50%",
          }
        }
      />
      <View
        padding={5}
      />
      <Input
        autoFocus={false}
        onChangeText={[Function]}
        placeholder="Email"
        style={
          Object {
            "borderColor": "#553555",
            "borderRadius": 10,
            "borderWidth": 1,
            "padding": 10,
            "width": "50%",
          }
        }
      />
      <View
        padding={5}
      />
      <Input
        onChangeText={[Function]}
        placeholder="Password"
        secureTextEntry={true}
        style={
          Object {
            "borderColor": "#553555",
            "borderRadius": 10,
            "borderWidth": 1,
            "padding": 10,
            "width": "50%",
          }
        }
      />
      <View
        padding={15}
      />
      <Button
        style={
          Object {
            "width": "50%",
          }
        }
      >
        Signup
      </Button>
      <View
        padding={5}
      />
      <TouchableOpacity
        activeOpacity={0.2}
      >
        <Poppins>
          Already have an account? Click here to login
        </Poppins>
      </TouchableOpacity>
    </View>
  </ScrollViewMock>
</KeyboardAvoidingView>
`;
